heimlig/src/integration/raw_jobs.rs

```diff
@@ -0,0 +1,967 @@
+                key_data,
+                key_size,
+                nonce_data,
+                nonce_size,
+                plaintext_data,
+                plaintext_size,
+                aad_data,
+                aad_size,
+                tag_data,
+                tag_size,
+            } => Request::EncryptChaChaPolyExternalKey {
+                client_id: client_id.into(),
+                request_id: request_id.into(),
+                key: check_pointer_and_size(key_data, key_size, &validator)?,
+                nonce: check_pointer_and_size(nonce_data, nonce_size, &validator)?,
+                plaintext: check_mut_pointer_and_size(plaintext_data, plaintext_size, &validator)?,
+                aad: check_pointer_and_size(aad_data, aad_size, &validator)?,
+                tag: check_mut_pointer_and_size(tag_data, tag_size, &validator)?,
+            },
+            RequestRaw::DecryptChaChaPoly {
+                client_id,
+                request_id,
+                key_id,
+                nonce_data,
+                nonce_size,
+                ciphertext_data,
+                ciphertext_size,
+                aad_data,
+                aad_size,
+                tag_data,
+                tag_size,
+            } => {
+                check_pointer_and_size(nonce_data, nonce_size, &validator)?;
+                check_pointer_and_size(ciphertext_data, ciphertext_size, &validator)?;
+                check_pointer_and_size(aad_data, aad_size, &validator)?;
+                check_pointer_and_size(tag_data, tag_size, &validator)?;
+                Request::DecryptChaChaPoly {
+                    client_id: client_id.into(),
+                    request_id: request_id.into(),
+                    key_id: key_id.into(),
+                    nonce: &[],
+                    ciphertext: &mut [],
+                    aad: &[],
+                    tag: &[],
+                }
+            }
+            RequestRaw::DecryptChaChaPolyExternalKey {
+                client_id,
+                request_id,
+                key_data,
+                key_size,
+                nonce_data,
+                nonce_size,
+                ciphertext_data,
+                ciphertext_size,
+                aad_data,
+                aad_size,
+                tag_data,
+                tag_size,
+            } => {
+                check_pointer_and_size(key_data, key_size, &validator)?;
+                check_pointer_and_size(nonce_data, nonce_size, &validator)?;
+                check_pointer_and_size(ciphertext_data, ciphertext_size, &validator)?;
+                check_pointer_and_size(aad_data, aad_size, &validator)?;
+                check_pointer_and_size(tag_data, tag_size, &validator)?;
+                Request::DecryptChaChaPolyExternalKey {
+                    client_id: client_id.into(),
+                    request_id: request_id.into(),
+                    key: check_pointer_and_size(key_data, key_size, &validator)?,
```

Comment:
Do we need to save the result in the struct?

Reply:
Yes, because decryption is done in-place.